src/main/perl/Fetch/Download.pm

```diff
@@ -94,17 +94,29 @@ sub retrieve
     my $ua = LWP::UserAgent->new();
     my $rq = HTTP::Request->new(GET => $url);
 
-    my $ht = scalar(localtime($time));
-    if (!$self->{FORCE}) {
+    # If human readable time ($ht) is not defined, then treat a 304 repsonse as an error
+    my $ht;
+    if ($self->{FORCE}) {
+        $self->verbose("FORCE set, not setting if_modified_since in request");
+    } elsif (! defined($time)) {
+        $self->verbose("modification time not defined, not setting if_modified_since in request");
+    } else {
+        $ht = scalar(localtime($time));
         $self->debug(1, "Retrieve if newer than $ht");
         $rq->if_modified_since($time);
     }
+
     $ua->timeout($self->{GET_TIMEOUT});
     $rq->header("Accept-Encoding" => join(" ", qw(gzip x-gzip x-bzip2 deflate)));
     my $rs = $ua->request($rq);
     if ($rs->code() == 304) {
-        $self->verbose("No changes on $url since $ht");
-        return 0;
+        if (defined($ht)) {
+            $self->verbose("No changes on $url since $ht");
+            return 0;
+        } else {
+            $self->warn("Unexpected no changes on $url without if_modified_since set in request");
```

Comment:
Nitpick: but this doesn't match the comment of line 97. If we get back a 304 then we get no data so we can't continue, so I think this is an error and the comment is correct. I think the message should be:
```
Server responded with code 304 for $url even though if_modified_since was not set in request. No profile retrieved.
```